gslib/utils/text_util.py

```diff
@@ -408,3 +409,23 @@ def get_random_ascii_chars(size, seed=0):
   contents = six.ensure_binary(contents)
   random.seed()  # Reset the seed for any other tests.
   return contents
+
+def get_prefix():
+  """Get a prefix string 'aaa-' for bucket names if using XML API
+
+  Since XML API doesn't support searching for bucket name prefixes, and since it
+  also only returns the first 1,000 buckets alphabetically, we sometimes see integration
+  tests flake when more than 1,000 buckets exist.
+
+  This creative workaround makes sure that buckets use in tab completion and wildcard tests
+  are hoisted to the top of the list of buckets. This workaround is primarily for our own CI
+  and tests may still flake if you have more than 1,000 buckets that would be ordered before
+  'aaa-' and are using the XML API.
+
+  Returns:
+    String literal 'aaa-' if currently using XML API.
+  """
+  if USING_JSON_API:
```

Comment:
I think it's fine to always return 'aaa-', even for the JSON API, in favor of not adding test logic to non-test modules.  (If you disagree, you could add this logic to the callsites in the test code instead)

Reply:
I'd be fine removing this method and doing this for the JSON API too! Applying it to the XML API only was trying to make sure it would take longer to hit >1000 buckets with `aaa-` prepended to them, but given the small set of tests this applies to, and the specific nature of this bug and workaround, going the simpler route and doing this regardless of API probably makes sense. :)